### PR TITLE
Qbft sync

### PR DESCRIPTION
**Pull request by alonmuroch**
A simple mockup of when qbft syncing is triggered

### CODE REVIEW

File: qbft/controller.go [hunk 1]
```diff
@@ -40,13 +40,15 @@ type Controller struct {
...
 	StoredInstances InstanceContainer
-	Domain          types.DomainType
-	Share           *types.Share
-	signer          types.SSVSigner
-	valueCheck      ProposedValueCheckF
-	storage         Storage
-	network         Network
-	proposerF       ProposerF
+	// HigherReceivedMessages holds all msgs from a higher height
+	HigherReceivedMessages *MsgContainer
```
Comment (nivBlox) [inline qbft/controller.go:44]: @alonmuroch missing initial for container

File: qbft/sync.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/pkg/errors"
+)
+
+func (c *Controller) processHigherHeightMsg(msg *SignedMessage) error {
+	added, err := c.HigherReceivedMessages.AddIfDoesntExist(msg)
+	if err != nil {
+		return errors.Wrap(err, "could not add higher height msg")
+	}
+	if added && c.f1SyncTrigger() {
+		// TODO should reset msg container? past msgs? all msgs?
```
Comment (nivBlox) [inline qbft/sync.go:14]: i think we can reset each time sync is done

Comment (nivBlox) [reply qbft/sync.go:14]: @alonmuroch also we need to add "clean" func to container

Comment (alonmuroch) [reply qbft/sync.go:14]: suggestions? to prevent replay attacks?

Comment (nivBlox) [reply qbft/sync.go:14]: clean once got f+1.
protect against replay attack will be handle using scoring as we discussed.

Comment (alonmuroch) [reply qbft/sync.go:14]: Not sure this should be "part of the spec" though..

**Pull request merged**